java/code/src/com/suse/oval/OVALCachingFactory.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.oval;
+
+import com.redhat.rhn.common.db.datasource.CallableMode;
+import com.redhat.rhn.common.db.datasource.DataResult;
+import com.redhat.rhn.common.db.datasource.ModeFactory;
+import com.redhat.rhn.common.db.datasource.Row;
+import com.redhat.rhn.common.db.datasource.SelectMode;
+import com.redhat.rhn.common.hibernate.HibernateFactory;
+
+import com.suse.oval.manager.OVALLookupHelper;
+import com.suse.oval.ovaltypes.DefinitionType;
+import com.suse.oval.ovaltypes.OvalRootType;
+import com.suse.oval.vulnerablepkgextractor.ProductVulnerablePackages;
+import com.suse.oval.vulnerablepkgextractor.VulnerablePackage;
+import com.suse.oval.vulnerablepkgextractor.VulnerablePackagesExtractor;
+import com.suse.oval.vulnerablepkgextractor.VulnerablePackagesExtractors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OVALCachingFactory extends HibernateFactory {
+    private static final Logger LOG = LogManager.getLogger(OVALCachingFactory.class);
+
+    private OVALCachingFactory() {
+        // Left empty on purpose
+    }
+
+    /**
+     * Extracts and save the list of vulnerable packages from {@code rootType}
+     *
+     * @param rootType the OVAL root to extract from
+     * */
+    // TODO: Following the single responsibility principle this method shouldn't have to extract vulnerable packages
+    //  itself
+    public static void savePlatformsVulnerablePackages(OvalRootType rootType) {
```

Comment:
I may be missing something in here, but where is this method called outside the unit tests?

Reply:
I think I found the reason for not being called outside the tests. Is it because the download and automatic extract are not yet implemented?